compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -734,22 +734,28 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                 if !impl_candidates.is_empty() && e.span.contains(span)
                     && let Some(expr) = exprs.first()
                     && let ExprKind::Path(hir::QPath::Resolved(_, path)) = expr.kind
-                    && let [path_segment] = path.segments
+                    && let [_] = path.segments
                 {
+                    let mut eraser = TypeParamEraser(self.tcx);
```

Comment:
Not sure if this is overkill, but we could know exactly what generic types need to be erased if we returned the impl's `DefId` from `find_similar_impl_candidates` instead of just the `TraitRef`, then we might just be able to use subst instead of erasing all type params

Reply:
yeah, but it might allow us to erase fewer type variables when we're calling one generic method from another generic method with type variables.

but probably not worth the effort...